CycloneDX/Services/GithubService.cs

```diff
@@ -27,54 +28,55 @@ namespace CycloneDX.Services
 
     public interface IGithubService
     {
-        Task<License> GetLicenseAsync(string licenseUrl, string version);
+        Task<License> GetLicenseAsync(string licenseUrl);
     }
 
-    class GithubService : IGithubService
+    public class GithubService : IGithubService
     {
 
-        private string _baseUrl;
+        private string _baseUrl = "https://api.github.com/";
         private HttpClient _httpClient;
-        private Regex _repositoryIdRegex = new Regex(@"^https?\:\/\/((github\.com)|(raw\.githubusercontent\.com))\/(?<repositoryId>[^\/]+\/[^\/]+)\/(blob\/)?master\/LICENSE.*$");
+        private List<Regex> _githubRepositoryRegexes = new List<Regex>
+        {
+            new Regex(@"^https?\:\/\/github\.com\/(?<repositoryId>[^\/]+\/[^\/]+)\/blob\/(?<refSpec>[^\/]+)\/LICENSE(.md)?$"),
+            new Regex(@"^https?\:\/\/raw\.githubusercontent\.com\/(?<repositoryId>[^\/]+\/[^\/]+)\/(?<refSpec>[^\/]+)\/LICENSE(.md)?$"),
```

Comment:
I'm not 100% sure about this, but I think I encountered some `LICENSE.txt` files in the past, so we may to allow some other extensions too.

Reply:
Interestingly the documentation states it needs to be `LICENSE` or `LICENSE.md`. But it also supports `LICENSE.txt`. Good pick up @ST-Apps.

But what doesn't work, unfortunately, is the ref parameter. Although it's supported, and the response includes the ref in returned urls, the license object is whatever is in master.

I've raised a support request with github about this. But in the meantime I think we'll just return license information when master is referenced. I think that will cover off almost all cases.